src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Extensions/CompilationExtensions.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+namespace Microsoft.CodeAnalysis.Shared.Extensions
+{
+    internal static class CompilationExtensions
+    {
+        public static INamedTypeSymbol? GetBestTypeByMetadataName(this Compilation compilation, string fullyQualifiedMetadataName)
```

Comment:
👍   I fel like this is written out in some other feature... i wish i could rememer where though so you could fix it up! :(

It might be 'generate HashCode' where we look for System.HashCode

Reply:
It's used heavily (exclusively?) in roslyn-analyzers. I can check for similar here.

Reply:
I wasn't able to find the other implementation. I'll file a follow-up issue to make `GetTypeByMetadataName` a banned API and switch everything to the new implementation.

Reply:
➡️ #43341